### ISSUE TITLE
'global.addEventListener is not a function' error during building react-native project

### ISSUE THREAD

**Issue by igorarkhipenko** [SocketCluster/socketcluster#329]
# Bug report

## Expected behavior: 
Adding of event listener to global object should be skipped

## Actual behaviour:
There is an error result assigned to variable during building in react-native app environment.
`var isBrowser = typeof window != 'undefined'` returns true which leads to trying to adding an event listener with

```
if (isBrowser && this.disconnectOnUnload) {
    this._unloadHandler = function () {
      self.disconnect();
    };

    global.addEventListener('beforeunload', this._unloadHandler, false); //right there an error is occured
  }
```

<img width="448" alt="screen shot 2017-09-22 at 9 57 37 am" src="https://user-images.githubusercontent.com/24496348/30732880-3a358af0-9f7d-11e7-8b69-0cec8617d4ad.png">

Here's a fragment from my`package.json`:
```
 "react-native": "0.46.4",
 "socketcluster": "^7.0.2",
 "socketcluster-client": "^7.0.1",
```

However, everything seems to work fine with changing 'window' to 'document' when browserity of environment is checked.

**Comment by jondubois** [comment]
@igorarkhipenko I tested it on the latest react native (for Android) and was able to reproduce this issues. I did not notice it before; maybe it's an issue with new versions of RN. I can't seem to find any info online about `window.addEventListener` being undefined.

**Comment by jondubois** [comment]
I'll push a fix which will check the existence of `window.addEventListener` - This event is mostly useful for browsers where a user might navigate to a different website - It's not usually relevant for WebView-based apps. It may be worthwhile to check on RN repo to see if anyone else has encountered a similar issues. It's not the first time that this kind of environment-detection logic has caused us problems with RN.

**Comment by jondubois** [comment]
@igorarkhipenko Fixed in `socketcluster-client@7.0.2`.

**Comment by igorarkhipenko** [comment]
@jondubois thank you for the fast response, it was a confusing moment for me too.
